test/testenv/deployment.go

```diff
@@ -562,3 +562,143 @@ func (d *Deployment) DeployLicenseMasterWithGivenSpec(name, licenseMasterName st
 	}
 	return deployed.(*enterprisev1.LicenseMaster), err
 }
+
+// DeploySingleSiteClusterWithGivenAppFrameworkSpec deploys indexer cluster (lm, shc optional) with app framework spec
+func (d *Deployment) DeploySingleSiteClusterWithGivenAppFrameworkSpec(name string, indexerReplicas int, shc bool, appFrameworkSpec enterprisev1.AppFrameworkSpec) error {
+
+	var licenseMaster string
+
+	// If license file specified, deploy License Master
+	if d.testenv.licenseFilePath != "" {
+		// Deploy the license master
+		_, err := d.DeployLicenseMaster(name)
+		if err != nil {
+			return err
+		}
+
+		licenseMaster = name
+	}
+
+	// Deploy the cluster master
+	cmSpec := enterprisev1.ClusterMasterSpec{
+		CommonSplunkSpec: enterprisev1.CommonSplunkSpec{
+			Spec: splcommon.Spec{
+				ImagePullPolicy: "Always",
+			},
+			Volumes: []corev1.Volume{},
+		},
+		AppFrameworkConfig: appFrameworkSpec,
+	}
+	_, err := d.DeployClusterMasterWithGivenSpec(name, licenseMaster, "", cmSpec)
+	if err != nil {
+		return err
+	}
+
+	// Deploy the indexer cluster
+	_, err = d.DeployIndexerCluster(name+"-idxc", licenseMaster, indexerReplicas, name, "")
+	if err != nil {
+		return err
+	}
+
+	// Deploy the SH cluster
+	shSpec := enterprisev1.SearchHeadClusterSpec{
+		CommonSplunkSpec: enterprisev1.CommonSplunkSpec{
+			Spec: splcommon.Spec{
+				ImagePullPolicy: "Always",
+			},
+			Volumes: []corev1.Volume{},
+		},
+		Replicas:           3,
+		AppFrameworkConfig: appFrameworkSpec,
+	}
+	if shc {
+		_, err = d.DeploySearchHeadClusterWithGivenSpec(name+"-shc", name, licenseMaster, "", shSpec)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// DeployMultisiteClusterWithSearchHeadAndAppFramework deploys cluster-master, indexers in multiple sites (SHC LM Optional) with app framework spec
+func (d *Deployment) DeployMultisiteClusterWithSearchHeadAndAppFramework(name string, indexerReplicas int, siteCount int, appFrameworkSpec enterprisev1.AppFrameworkSpec, shc bool) error {
```

Comment:
As discussed, see if we can optimize by reusing existing methods with arguments as part of future stories